src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -129,6 +161,88 @@ private Uri GetAccountUri(AzureBlobEgressProviderOptions options, out string acc
             return blobUriBuilder.ToUri();
         }
 
+        private Uri GetQueueAccountUri(AzureBlobEgressProviderOptions options, out string accountName)
+        {
+            var queueUriBuilder = new QueueUriBuilder(options.QueueAccountUri);
+
+            queueUriBuilder.Query = null;
+            queueUriBuilder.QueueName = null;
+
+            accountName = queueUriBuilder.AccountName;
+
+            return queueUriBuilder.ToUri();
+        }
+
+        private async Task EgressMessageToQueue(string blobName, AzureBlobEgressProviderOptions options, CancellationToken token)
+        {
+            try
+            {
+                QueueClient queueClient = await GetQueueClientAsync(options, token);
+
+                if (queueClient.Exists())
+                {
+                    queueClient.SendMessage(blobName);
+                }
+                else
+                {
+                    Logger.LogWarning(string.Format(CultureInfo.CurrentCulture, Strings.Message_QueueDoesNotExist, options.QueueName));
+                }
+            }
+            catch (AggregateException ex) when (ex.InnerException is RequestFailedException innerException)
```

Comment:
Why does it matter that the inner exception is a `RequestFailedException`?

Reply:
I actually carried the exception handling logic over from the `EgressAsync` methods (i.e. Line 129), since I figured it was best to be consistent with what was already in place for egressing to blob storage. That being said, would you prefer something different?